GameServerLib/GameObjects/AttackableUnits/AI/Champion.cs

```diff
@@ -364,6 +364,23 @@ public void Respawn()
             RespawnTimer = -1;
         }
 
+        public bool HandleDisconnect()
+        {
+            var peerinfo = _game.PlayerManager.GetPeerInfo(this._playerId);
+            if (peerinfo != null)
+            {
+                if (!peerinfo.IsDisconnected)
+                {
+                    _game.PacketNotifier.NotifyUnitAnnounceEvent(UnitAnnounces.SUMMONER_DISCONNECTED, peerinfo.Champion);
+                }
+                peerinfo.IsDisconnected = true;
+                peerinfo.Champion.StopChampionMovement();
```

Comment:
Change `peerinfo.Champion` to `this` in all occurrences.  Also, do not use `peerInfo` and so here. Try to put all these things in `Game` class, to decouple Peer from the logic of the objects.

Also, do you know why `HandleDisconnect` should check if the peer is disconnected, does it really get called when the peer is not disconnected?

Reply:
I forgot to refactor that function, kill me. 

I thought that check maybe was to prevent double calling the packetnotifier, but I tested without and it worked fine so I removed it.